### DISCUSSION TITLE
ISSUES with getting stuck in drafts

### DISCUSSION THREAD

**Discussion by sharkmullet** [Extended-Thunder/send-later#500] [Q&A]
I don't know why they never leave the drafts folder. I tried repairing the drafts folder, and I also restarted my computer.

Messages don't leave the drafts folder after the time they were supposed to go.

What do i do?

**Comment by sharkmullet** [comment]
btw, i have version 92.4

**Comment by philfutureproof** [comment]
I'm also having issues where the emails appear in the draft folder but then doesn't leave there after the set time has passed.

Have restarted computer but what's this thing about 'repairing the drafts folder' - how do I do that? 

I'm on Mac: 10.13.6
Thunderbird: 102.6.0
Send Later: 9.2.4

**Comment by philfutureproof** [comment]
OK, have repaired the various draft folders but still no luck!

**Comment by damienjadeduff** [comment]
I have had the same problem recently.
SendLater had upgraded to 9.2.4.

Temporary fix:
Reverting to 9.0.11 fixed this.
You need to uninstall SendLater then download the older version from here https://addons.thunderbird.net/en-us/thunderbird/addon/send-later-3/versions/
And install via extensions > gear icon > Install Add-on from file...

In my case I have yet to do the requisite research to find out how to peg the version till a new one comes through.

**Reply by damienjadeduff** [reply]
Issue here: https://github.com/Extended-Thunder/send-later/issues/501

**Reply by philfutureproof** [reply]
Wonderful Damien - that works nicely for the moment. I guess the developer Jonathan Perry-Houts will pick up on the comversation and sort things out on a more long term basis. All the best

**Reply by damienjadeduff** [reply]
To make the downgrade stick disable "Update Add-ons Automatically"
<img width="607" alt="image" src="https://user-images.githubusercontent.com/4210278/208903352-9e7235a4-5675-45ba-b353-81f43d37a678.png">

**Comment by philfutureproof** [comment]
Good point!

Have done so

Phil



On 21/12/2022 12:18, Damien Jade Duff wrote:
>
> To make the downgrade stick disable "Update Add-ons Automatically"
> image 
> <https://user-images.githubusercontent.com/4210278/208903352-9e7235a4-5675-45ba-b353-81f43d37a678.png>
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/Extended-Thunder/send-later/discussions/500#discussioncomment-4467153>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AD3OSS5KVMXCWUCTSOUN4IDWOLYPVANCNFSM6AAAAAAS6ZHMQM>.
> You are receiving this because you commented.Message ID: 
> ***@***.***>
>

-- 

Phil Legg

MD/Founder

Mob: +44 (0)7956 514 897

Futureproof logo
*Futureproof Marketing & Promotions
*330 Westbourne Park Road
London W11 1EQ.
UK.

Tel +44 (0)20 7792 8597

www.futureproofpromotions.com <http://www.futureproofpromotions.com>
www.facebook.com/futureproofPR <http://www.facebook.com/futureproofPR>
www.twitter.com/futureproofPR <http://www.twitter.com/futureproofPR>
www.youtube.com/futureproofPR <http://www.youtube.com/futureproofPR>

E-MAIL DISCLAIMER: The information in this e-mail is confidential and 
may be legally privileged. It is intended solely for the addressee and 
access to the e-mail by anyone else is unauthorised. If you are not the 
intended recipient, any disclosure, copying, distribution or any action 
taken or omitted to be taken in reliance on it, is prohibited and may be 
unlawful. When addressed to our clients, any opinions or advice 
contained in this e-mail are subject to the terms and conditions 
expressed in the governing client engagement letter or contract.

**Comment by jobisoft** [comment]
Those of you affected, does it help if you use a drafts folder, which is not inside an IMAP folder, but inside a local folder?

**Reply by damienjadeduff** [reply]
@jobisoft In my case some of the IMAP accounts I seldom use indeed had drafts (account settings > account name > copies and folders > drafts and templates > Keep draft messages in) set in the IMAP accounts. However, setting them all to use local folders, upgrading, I had the same problem. Downgrading again fixed it.

**Reply by chris3392** [reply]
I am having the same problem for about 2 weeks.
my default is to use a draft in  local folder; the emails sit there, eventhough the "send later" column shows the expected send time

**Comment by jobisoft** [comment]
@damienjadeduff : Thanks for your feedback. Do you see any errors in the Javascript Error console while 
a) using send later to place the message in draft and/or 
b) while send later should be actually sending the message?

The error console is at Tools -> Developer Tools -> Error Console or CTRL + SHIFT + J

**Reply by damienjadeduff** [reply]
Regarding a) if it is helpful, the version that works (9.0.11) gives an unresolved promise or something error (I can supply if you wish). There seems to be no error for the latest version (9.2.4) however.

(a)
Latest version (9.2.4)
<img width="1423" alt="image" src="https://user-images.githubusercontent.com/4210278/209332805-6f59247b-6854-4aff-a955-719dbe7d7516.png">

For (b) and 9.2.4 there is nothing.

**Reply by damienjadeduff** [reply]
I do get these errors but on thunderbird start and using add-ons:

<img width="1437" alt="image" src="https://user-images.githubusercontent.com/4210278/209333164-a65a5857-222d-4893-a8d5-ada8340bc899.png">

**Reply by damienjadeduff** [reply]
Though I doubt that relevant?

**Comment by jobisoft** [comment]
This *could* indicate an issue with your profile. Is it an old one, upgraded from version to version, or a fresh one for 102?

If you use an IMAP account, you could create a new test profile, setup the IMAP account and try 9.2.4 there. I would not suggest this if you are using POP.

**Reply by damienjadeduff** [reply]
Yes in my case it could be just my old profile (yes it is old... reasons...) but because this is hitting a few people I thought it would likely not be just me. Let's wait for someone else to chime in. Thanks for your time on this.

**Comment by jobisoft** [comment]
It would help to know if all affected users are using older profiles and if a new profile fixes it for everybody.

**Comment by jph-sendlater** [comment]
Cross posting from the related issue: https://github.com/Extended-Thunder/send-later/issues/501#issuecomment-1364431725

I'm hoping that some performance optimizations in the next version (9.2.5) will solve this Please let me know once you've tried the new version whether this problem is persisting.

**Reply by damienjadeduff** [reply]
At least for me the problem persists. Downgrading to 9.0.11 sends the messages immediately.

**Reply by damienjadeduff** [reply]
It turns out the promise resolution errors just occur when finishing with the dialog too fast:
<img width="721" alt="image" src="https://user-images.githubusercontent.com/4210278/209536747-8e3bd978-ee56-468f-b66c-a6e23a575d9b.png">

(9.2.5).

Probably not related because I get these in 9.0.11 too.

**Comment by igzmanuelmartinvivaldi** [comment]
Same problem for me in latest version (Windows 11)
_Send Later version 9.2.7 on Thunderbird 102.6.1 (20221219215418) [win x86-64]_
It's 19:56 and no email is sent (the first date at the left of the image is the Send Later Column)
![image](https://user-images.githubusercontent.com/2234919/211637720-e943f0fa-e1e2-413b-bbc6-1bfa0fe10c4a.png)
No errors/warnings/messages in console when sending later or at the expected sending time
No warning when closing Thunderbird that some emails are not sent yet

**Comment by rsl0** [comment]
Hi, I'm not a techie so please excuse if I'm stating the obvious.

I have run into the same issue as notified above i.e. - Send later not delivering the messages. 

I'm using TBx64 102.6.1   - Win10. 

I've reverted to Send Later version 9.0.11 and can confirm it is working.

I only use POP accounts and my TB profile is quite old.

The one thing I noticed with Send later v9.2.4 & 9.2.7 is that the bottom status bar was missing the text, "Send later [IDLE]".

I have used Send later for a few years now and wanted to notify of this issue.

Thank you

**Comment by atscomms** [comment]
This is still an issue in V 9.2.7 w/ TB 102.8.0 (32-bit) on a Windows 10 64bit PC
Email stays in Drafts folder, not going out at specified time.
The past date & time appears in the Draft Folder's first column, and there is a numbered flag on the Send Later icon in the menu bar.

**Comment by mbaillybechet** [comment]
Hello,

got the same issue (and same fix: downgrade to version 9.0.11) to fix it.

I am on Linux Ubuntu 22.04, Thunderbird 102.7.1, and non working version was Send Later 9.2.7

Email stayed stuck in the draft folders (with good X-Send-Later in the header). Installation of version 9.0.11 immediatly sent all tests emails that were waiting :-)

**Comment by My1** [comment]
clean up your drafts possibly. if you have more than 100 mails, it seems to stop processing after 100.

**Comment by jikamens** [comment]
There are two possible causes of this issue of which I am unaware:

1. If you have more than 100 messages in a Drafts folder then Send Later may not detect all scheduled drafts in that folder. This is being tracked in #529. We have a fix, which will be included in the next Send Later release.
2. There is a race condition in the Send Later start-up code which occurs for some users when Thunderbird starts up, which prevents Send Later from initializing itself properly and subsequently prevents scheduled messages from being sent. We are also hopeful that this will be fixed in the next Send Later release, though we're not as certain about this fix as we are about the other one since the root cause of this issue is less clear.

You can tell if the race condition is impacting you if you don't see "Send Later" as one of the columns you are able to add to the message list in your Drafts folder.

**Comment by jikamens** [comment]
I have a new prerelease which I am hopeful does a much better job of fixing the issue of messages getting stuck in the Drafts folder:

https://github.com/Extended-Thunder/send-later/releases/tag/v10.0.3

Can those of you who are still having this problem give it a try?

To do that, download the send_later.xpi file from that release page (don't just click the link, right-click and select download so that the browser doesn't try to install it as an extension to itself), then open the Add-ons page in Thunderbird, click the gear icon, select "Install Add-on From File", and browse to the downloaded XPI file. Please comment here on whether or not it improves things for you.

**Comment by jikamens** [comment]
I'm going to close this discussion because these are really bugs, not discussion topic. If you are still having problems of this sort please [open a new issue](https://github.com/Extended-Thunder/send-later/issues/new) about them.
